controllers/machineset_controller.go

```diff
@@ -381,28 +373,15 @@ func (r *MachineSetReconciler) syncReplicas(ctx context.Context, ms *clusterv1.M
 		// Choose which Machines to delete.
 		machinesToDelete := getMachinesToDeletePrioritized(machines, diff, deletePriorityFunc)
 
-		errCh := make(chan error, diff)
-		var wg sync.WaitGroup
-		wg.Add(diff)
-		for _, machine := range machinesToDelete {
-			go func(targetMachine *clusterv1.Machine) {
-				defer wg.Done()
-				err := r.Client.Delete(context.Background(), targetMachine)
-				if err != nil {
-					logger.Error(err, "Unable to delete Machine", "machine", targetMachine.Name)
-					r.recorder.Eventf(ms, corev1.EventTypeWarning, "FailedDelete", "Failed to delete machine %q: %v", targetMachine.Name, err)
-					errCh <- err
-				}
-				logger.Info("Deleted machine", "machine", targetMachine.Name)
-				r.recorder.Eventf(ms, corev1.EventTypeNormal, "SuccessfulDelete", "Deleted machine %q", targetMachine.Name)
-			}(machine)
-		}
-		wg.Wait()
-		close(errCh)
-
 		var errs []error
-		for err := range errCh {
-			errs = append(errs, err)
+		for _, machine := range machinesToDelete {
+			if err := r.Client.Delete(context.Background(), machine); err != nil {
+				logger.Error(err, "Unable to delete Machine", "machine", machine.Name)
+				r.recorder.Eventf(ms, corev1.EventTypeWarning, "FailedDelete", "Failed to delete machine %q: %v", machine.Name, err)
+				errs = append(errs, err)
+			}
```

Comment:
Am I wrong or there should be a continue before closing the if?

Reply:
You're not! Good catch, guess we should have had that before as well